File: 백준/Silver/7795. 먹을 것인가 먹힐 것인가/먹을 것인가 먹힐 것인가.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder stb = new StringBuilder();
        int t = Integer.parseInt(br.readLine());
        String[] input;

        while (t-- > 0) {
            input = br.readLine().split(" ");
            int n = Integer.parseInt(input[0]);
            int m = Integer.parseInt(input[1]);
            int[] arrA = new int[n];
            int[] arrB = new int[m];

            input = br.readLine().split(" ");
            for(int i = 0; i < n; i++) {
                arrA[i] = Integer.parseInt(input[i]);
            }

            input = br.readLine().split(" ");
            for(int i = 0; i < m; i++) {
                arrB[i] = Integer.parseInt(input[i]);
            }

            Arrays.sort(arrB);
            int cnt = 0;

            for (int i = 0; i < n; i++) {
                int start = 0, end = m - 1;

                while (start <= end) {
                    int mid = (start + end) / 2;

                    if (arrB[mid] < arrA[i]) {
                        start = mid + 1;
                    } else {
                        end = mid - 1;
                    }
                }

                cnt += start;
            }

            stb.append(cnt).append("\n");
        }

        System.out.println(stb.toString());
    }
}

File: 백준/Silver/7795. 먹을 것인가 먹힐 것인가/먹을 것인가 먹힐 것인가.js
const input = require("fs")
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "input.txt")
  .toString()
  .trim()
  .split("\n");

let t = Number(input.shift());
let ans = [];

while (t-- > 0) {
  let [n, m] = input.shift().split(" ").map(Number);
  let arrA = input.shift().split(" ").map(Number);
  let arrB = input.shift().split(" ").map(Number);

  arrB.sort((a, b) => a - b);

  let cnt = 0;
  arrA.map((v) => {
    let start = 0,
      end = m - 1;

    while (start <= end) {
      let mid = Math.floor((start + end) / 2);
      if (arrB[mid] < v) {
        start = mid + 1;
      } else {
        end = mid - 1;
      }
    }

    cnt += start;
  });

  ans.push(cnt);
}

console.log(ans.join("\n"));
